stripe/src/main/java/com/stripe/android/googlepay/StripeGooglePayLauncher.kt

```diff
@@ -0,0 +1,109 @@
+import kotlinx.parcelize.Parceler
+import kotlinx.parcelize.Parcelize
+
+internal class StripeGooglePayLauncher : ActivityStarter<StripeGooglePayActivity, StripeGooglePayLauncher.Args> {
+    constructor(activity: Activity) : super(
+        activity,
+        StripeGooglePayActivity::class.java,
+        REQUEST_CODE,
+        intentFlags = Intent.FLAG_ACTIVITY_NO_ANIMATION
+    )
+
+    constructor(fragment: Fragment) : super(
+        fragment,
+        StripeGooglePayActivity::class.java,
+        REQUEST_CODE,
+        intentFlags = Intent.FLAG_ACTIVITY_NO_ANIMATION
+    )
+
+    @Parcelize
+    data class Args(
+        internal var paymentIntent: PaymentIntent,
+
+        /**
+         * ISO 3166-1 alpha-2 country code where the transaction is processed.
+         */
+        internal var countryCode: String,
+
+        /**
+         * Set to true to request an email address.
+         */
+        internal var isEmailRequired: Boolean = false
+    ) : ActivityStarter.Args {
+
+        companion object {
+            @JvmSynthetic
+            internal fun create(intent: Intent): Args {
+                return requireNotNull(intent.getParcelableExtra(ActivityStarter.Args.EXTRA))
+            }
+        }
+    }
+
+    sealed class Result : ActivityStarter.Result {
+        override fun toBundle(): Bundle {
+            return Bundle().also {
+                it.putParcelable(ActivityStarter.Result.EXTRA, this)
+            }
+        }
+
+        @Parcelize
+        class Error(
+            val exception: Throwable,
+            val googlePayStatus: Status? = null,
+            val paymentMethod: PaymentMethod? = null,
+            val shippingInformation: ShippingInformation? = null
+        ) : Result() {
+            companion object : Parceler<Error> {
+                override fun create(parcel: Parcel): Error {
+                    return Error(
+                        exception = parcel.readSerializable() as Throwable,
+                        googlePayStatus = parcel.readParcelable(Status::class.java.classLoader)
+                    )
+                }
+
+                override fun Error.write(parcel: Parcel, flags: Int) {
+                    parcel.writeSerializable(exception)
+                    parcel.writeParcelable(googlePayStatus, flags)
+                }
+            }
+        }
+
+        @Parcelize
+        class PaymentIntent(
+            val paymentIntentResult: PaymentIntentResult
+        ) : Result()
+
+        @Parcelize
+        object Canceled : Result()
+
+        @Parcelize
+        object Unavailable : Result()
+
+        companion object {
+            /**
+             * @return the [Result] object from the given `Intent`
+             */
+            @JvmStatic
+            fun fromIntent(intent: Intent): Result? {
+                return intent.getParcelableExtra(ActivityStarter.Result.EXTRA)
+            }
+        }
+    }
+
+    companion object {
+        const val REQUEST_CODE: Int = 9004
```

Comment:
out of curoisity why 9004?

Reply:
No particular reason, but we should investigate a better strategy for request code management.